hamilton/caching/adapter.py

```diff
@@ -1173,7 +1265,23 @@ def do_node_execute(
                 CachingBehavior.IGNORE,
             ):
                 cache_key = self.get_cache_key(run_id=run_id, node_name=node_name, task_id=task_id)
+
+                # nodes collected in `._data_loaders` return tuples of (result, metadata)
+                # where metadata often includes a timestamp. To ensure we provide a consistent
+                # `data_version` / hash, we only hash the result part of the materializer return
+                # value and discard the metadata.
+                if node_name in self._data_loaders[run_id] and isinstance(result, tuple):
+                    result = result[0]
+
                 data_version = self._version_data(node_name=node_name, run_id=run_id, result=result)
+
+                # nodes collected in `._data_savers` return a dictionary of metadata
```

Comment:
I'm not convinced we need to special case these... But for now, let's call this a bit experimental (data savers/loaders) -- it's a very odd case that we don't want ot dwell onmore.